common/string_util.c

```diff
@@ -101,6 +101,12 @@ char *strstr(const char *haystack, const char *needle)
    return NULL;
 }
 
+#if !defined(TESTING) && !defined(USERMODE_APP)
+   char *strstr(const char *haystack, const char *needle) \
```

Comment:
Good job here. Just, I don’t understand why you did that only for strstr() and not for the other functions. See the other comments.